tests/src/compiler/hipClassKernel.h

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright (c) 2015-2017 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+#ifndef COMPILER_HIPCLASSKERNEL_H_
+#define COMPILER_HIPCLASSKERNEL_H_
+
+#include <iostream>
+#include "hip/hip_runtime.h"
+#include "test_common.h"
+
+static const int BLOCKS = 512;
+static const int THREADS_PER_BLOCK = 1;
+static const int ENABLE_DESTRUCTOR_TEST = 0;
```

Comment:
Is this test fails for both hcc and hip-clang? If not, should differentiate hcc and hip-clang by using macro __HCC__

Reply:
Hi @yxsamliu , Yes. the test fails with both HCC and HIP-Clang, but with different options. Will enable it for HIP-Clang alone when the noopts issue is fixed.